loopy/codegen/__init__.py

```diff
@@ -480,6 +480,20 @@ def generate_code_for_a_single_kernel(kernel, callables_table, target,
 
     codegen_plog = ProcessLogger(logger, f"{kernel.name}: generate code")
 
+    # {{{ pre-codegen-process of non-entrypoint kernel
+
+    if not is_entrypoint:
+        from loopy.kernel.array import ArrayBase
+        from loopy.kernel.data import auto
+
+        new_args = [arg.copy(offset=0 if arg.offset is auto else arg.offset)
+                    if isinstance(arg, ArrayBase)
+                    else arg
+                    for arg in kernel.args]
+        kernel = kernel.copy(args=new_args)
```

Comment:
Can this be done without modifying things? I think it's not a good practice. I would prefer to remove the effect, or if that's hard, maybe `arg.offset is auto` should be an error for callable kernels.

Reply:
> I would prefer to remove the effect,

I.e. checking everywhere where we emit offset code and jumping over it for non-entrypoint kernels?

Reply:
> Can this be done without modifying things? I think it's not a good practice.

FWIW, I tend to agree with this sentiment, just that the other option was turning out to be slightly disruptive so I took this route.

Reply:
In a way, this highlights that this `offset` thing has leaked too far into the IR. It's a concern that's specific to the OpenCL entrypoints and nobody else, so it's kind of silly that any other target even needs to know about it. #432

In the meantime, I see the difficulty you describe. Maybe this is survivable until we remove it. Let's find out.